### PR TITLE
Support Bucket Replication

### PR DESCRIPTION
**Pull request by goruha**
## What
* Added ability to grant sufficient IAM permissions to accept replication from master bucket

## Why
* We want s3 bucket to be the destination in replication process

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ module "website_with_cname" {
 | `logs_standard_transition_days`     | `30`           | Number of days to persist in the standard storage tier before moving to the glacier tier                        | No       |
 | `logs_glacier_transition_days`      | `60`           | Number of days after which to move the data to the glacier storage tier                                         | No       |
 | `logs_expiration_days`              | `90`           | Number of days after which to expunge the objects                                                               | No       |
-
+| `masters`                           | `{}`           | Source bucket name and AWS IAM ID of owner for replication. In format map { "$id" = "$bucket_name" }            | No       |
```
Comment (osterman) [inline README.md]: What is the `$id`?

Comment (osterman) [reply README.md]: let's rename what you call `$id` to `$aws_account_id`

Comment (osterman) [reply README.md]: Should we invert the map? Bucket names are globally unique, but multiple buckets might exist in the same account. To permit multiple masters in the same account, we should invert.

File: main.tf [hunk 2]
```diff
@@ -80,6 +81,31 @@ data "aws_iam_policy_document" "default" {
...
+
+    actions = [
+      "s3:GetBucketVersioning",
+      "s3:PutBucketVersioning",
+      "s3:ReplicateObject",
+      "s3:ReplicateDelete",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.default.arn}",
```
Comment (osterman) [inline main.tf]: In the original version, this was the source bucket arn. Now this is the destination bucket arn.

Comment (osterman) [reply main.tf]: https://github.com/cloudposse/terraform-aws-s3-website/pull/10/commits/96450a37a9b1909b855f0ad1ed120690493d7025#diff-7a370d8342e7203b805911c92454f0f4R105

File: main.tf [hunk 3]
```diff
@@ -80,6 +81,31 @@ data "aws_iam_policy_document" "default" {
       type        = "AWS"
       identifiers = ["*"]
     }
+  }]
+
+  statement = ["${flatten(data.aws_iam_policy_document.master.*.statement)}"]
+}
+
+data "aws_iam_policy_document" "master" {
```
Comment (osterman) [inline main.tf]: I can't see where you associate this policy with the bucket.

Also, heads up: https://github.com/hashicorp/terraform/issues/10543#issuecomment-265657860

File: main.tf (hunk 2 above)
Comment (goruha) [reply main.tf]: It was mistake

File: main.tf (hunk 3 above)
Comment (goruha) [reply main.tf]: Here it is https://github.com/cloudposse/terraform-aws-s3-website/pull/10#discussion-diff-141544489R86
We do not attach policy, we get array of statements and include it to default policy document

File: README.md [hunk 4]
```diff
@@ -68,6 +68,7 @@ module "website_with_cname" {
 | `logs_standard_transition_days`     | `30`           | Number of days to persist in the standard storage tier before moving to the glacier tier                        | No       |
 | `logs_glacier_transition_days`      | `60`           | Number of days after which to move the data to the glacier storage tier                                         | No       |
 | `logs_expiration_days`              | `90`           | Number of days after which to expunge the objects                                                               | No       |
+| `master_aws_account_ids             | `[]`           | List of AWS IAM ID of owner for replication source bucket.                                                      | No       |
```
Comment (osterman) [inline README.md]: I prefer we use ARNs for consistency. See `deployment_arns`

**Pull request merged**